backend/operations_scanner_test.go

```diff
@@ -74,20 +77,26 @@ func TestDeleteOperationCompleted(t *testing.T) {
 			defer server.Close()
 
 			scanner := &OperationsScanner{
-				dbClient:           database.NewCache(),
+				dbClient:           mockDBClient,
 				notificationClient: server.Client(),
 			}
 
 			operationDoc := database.NewOperationDocument(database.OperationRequestDelete, resourceID, internalID)
 			operationDoc.NotificationURI = server.URL
 			operationDoc.Status = tt.operationStatus
 
-			_ = scanner.dbClient.CreateOperationDoc(ctx, operationDoc)
+			var resourceDocDeleted bool
 
-			if tt.resourceDocPresent {
-				resourceDoc := database.NewResourceDocument(resourceID)
-				_ = scanner.dbClient.CreateResourceDoc(ctx, resourceDoc)
-			}
+			mockDBClient.EXPECT().
+				DeleteResourceDoc(gomock.Any(), resourceID).
+				Do(func(ctx context.Context, resourceID *arm.ResourceID) {
+					resourceDocDeleted = tt.resourceDocPresent
+				})
+			mockDBClient.EXPECT().
+				UpdateOperationDoc(gomock.Any(), operationDoc.ID, gomock.Any()).
+				DoAndReturn(func(ctx context.Context, operationID string, callback func(*database.OperationDocument) bool) (bool, error) {
```

Comment:
`Return` instead?

Reply:
I don't think so here, because we have to invoke the callback that's passed in as a function argument.